--- ./CMSSW_CVS/DataFormats/TrackingRecHit/src/RecHit2DLocalPos.cc	2007-08-10 18:24:54.000000000 -0400
+++ RecHit2DLocalPos.cc	2013-03-14 13:41:03.000000000 -0400
@@ -1,14 +1,5 @@
 #include "DataFormats/TrackingRecHit/interface/RecHit2DLocalPos.h"
 
-void RecHit2DLocalPos::initialize() const
-{
-  theProjectionMatrix = AlgebraicMatrix( 2, 5, 0);
-  theProjectionMatrix[0][3] = 1;
-  theProjectionMatrix[1][4] = 1;
-
-  isInitialized = true;
-}
-
 AlgebraicVector RecHit2DLocalPos::parameters() const 
 {
   AlgebraicVector result(2);
@@ -38,9 +29,10 @@
   return nullvector; 
 }
 
-
-
-// static member definition
-
-bool RecHit2DLocalPos::isInitialized( false);
-AlgebraicMatrix RecHit2DLocalPos::theProjectionMatrix;
+AlgebraicMatrix RecHit2DLocalPos::projectionMatrix() const
+{
+  AlgebraicMatrix m( 2, 5, 0);
+  m[0][3] = 1;
+  m[1][4] = 1;
+  return m;
+}
